crates/accelerate/src/sabre_layout.rs

```diff
@@ -114,15 +114,16 @@ fn layout_trial(
     max_iterations: usize,
     num_swap_trials: usize,
     run_swap_in_parallel: bool,
-) -> (NLayout, Vec<usize>, SabreResult) {
-    let num_physical_qubits = distance_matrix.shape()[0];
+) -> (NLayout, Vec<PhysicalQubit>, SabreResult) {
+    let num_physical_qubits: u32 = distance_matrix.shape()[0].try_into().unwrap();
```

Comment:
This is functionally equivalent to `as u32` right as both will just panic if it's too large for a u32? Is there a difference in the generated code between this vs the straight cast?

Reply:
`as` casts silently truncate rather than panicking.